src/components/List/HierarchyList/HierarchyList.jsx

```diff
@@ -87,25 +87,98 @@ export const searchItem = (item, searchTerm) => {
   const filteredItems = [];
   cloneDeep(items).forEach(item => {
     // if the item has children, recurse and search children
     if (item.children) {
       // eslint-disable-next-line
-      item.children = searchNestedItems(item.children, value);
+      item.children = searchForNestedItemValues(item.children, value);
       // if it's children did, we still need the item
       if (item.children.length > 0) {
         filteredItems.push(item);
       }
     } // if the item matches, add it to the filterItems array
-    else if (searchItem(item, value)) {
+    else if (searchForItemValue(item, value)) {
       filteredItems.push(item);
     }
   });
   return filteredItems;
 };
 
+/**
+ * Searches an item for a specific id
+ * @param {Object} item to be searched
+ * @param {string} searchId
+ * @returns {Boolean} found or not
+ */
+export const searchForItemId = (item, searchId) => {
+  if (
+    // Check if the value or secondary value has a match
+    item.id.toLowerCase().search(searchId.toLowerCase()) !== -1
+  ) {
+    return true;
+  }
+  return false;
+};
+
+/**
+ * Assumes that the first level of items is not searchable and is only uses to categorize the
+ * items. Because of that, only search through the children. If a child is found while filtering,
+ * it needs to be returned to the filtered array. Deep clone is required because spread syntax is
+ * only a shallow clone
+ * @param {Array<Object>} items
+ * @param {String} value what to search for
+ * @returns {Array<Object>}
+ */
+export const searchForNestedItemIds = (items, value) => {
+  const filteredItems = [];
+  cloneDeep(items).forEach(item => {
+    // if the item has children, recurse and search children
+    if (item.children) {
+      // eslint-disable-next-line
+      item.children = searchForNestedItemIds(item.children, value);
+      // if it's children did, we still need the item
+      if (item.children.length > 0) {
+        filteredItems.push(item);
+      }
+    } // if the item matches, add it to the filterItems array
+    else if (searchForItemId(item, value)) {
+      filteredItems.push(item);
+    }
+  });
+  return filteredItems;
+};
+
+/**
+ * The ref should only be set if the item is selected by default. By using a callback,
+ * we can check if a node is properly attached to the ref, then scroll to it within the list
+ */
+const useHookWithRefCallback = () => {
+  const ref = useRef(null);
+  const setRef = useCallback(
+    node => {
+      if (node) {
+        // Check if a node is actually passed. Otherwise node would be null.
+        // node is the text of the item
+        // parentNode is the container div with the button role
+        // parentNode.parentNode is the list content, which is also the element to be scrolled
+        // the offsetHeight needs to be multiplied by 3 to be able to view the whole element
+        const offset =
+          node.offsetTop -
+          node.parentNode.offsetHeight -
```

Comment:
need to optionally null check these keys

Reply:
Fixed